missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/PlayerEquipmentRandomizer.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of MissileWars (https://github.com/Butzlabben/missilewars).
+ * Copyright (c) 2018-2021 Daniel Nägele.
+ *
+ * MissileWars is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MissileWars is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MissileWars.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.butzlabben.missilewars.util;
+
+import de.butzlabben.missilewars.Logger;
+import de.butzlabben.missilewars.game.Game;
+import de.butzlabben.missilewars.wrapper.abstracts.Arena;
+import de.butzlabben.missilewars.wrapper.missile.Missile;
+import de.butzlabben.missilewars.wrapper.player.MWPlayer;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Collections;
+
+/**
+ * @author Butzlabben
+ * @since 19.01.2018
+ */
+
+public class PlayerEquipmentRandomizer {
+
+    private final MWPlayer mwPlayer;
+    private final Game game;
+    private final Arena arena;
+
+    private final int maxGameDuration;
+    private final int defaultIntervalByTeamAmount = 20;
+    private final int defaultFactorByGameTime = 1;
+
+    int playerInterval;
+    int sendEquipmentCounter = 0;
+
+
+    public PlayerEquipmentRandomizer(MWPlayer mwPlayer, Game game) {
+        this.mwPlayer = mwPlayer;
+        this.game = game;
+        this.arena = game.getArena();
+        maxGameDuration = game.getArena().getGameDuration() * 60;
+
+        setPlayerInterval(getStartInterval() + 1);
+    }
+
+    public void nextPeriod() {
+
+        playerInterval--;
+        mwPlayer.getPlayer().setLevel(playerInterval);
+
+        if (playerInterval <= 0) {
+
+            sendRandomGameEquipment();
+            setPlayerInterval((int) Math.ceil(getIntervalByTeamAmount() * getFactorByGameTime()));
+        }
+
+    }
+
+    private void setPlayerInterval(Integer playerInterval) {
+        this.playerInterval = playerInterval;
+        mwPlayer.getPlayer().setLevel(playerInterval);
+    }
+
+    /**
+     * This method randomly shuffles the game equipment list and gives
+     * the player the first item of that list. The ratio is 2:1 between
+     * missiles and special items.
+     */
+    private void sendRandomGameEquipment() {
+
+        ItemStack item;
+
+        // switch between type of "items":
+        // after 2 missile items, you get one special item
+        if (sendEquipmentCounter >= 2) {
+
+            Collections.shuffle(game.getSpecialEquipment().getSpecialEquipmentList());
+            item = game.getSpecialEquipment().getSpecialEquipmentList().get(0);
```

Comment:
Don't shuffle the collection for every new item, rather generate a random number between 0 (including) and the length (excluding) of the collection and access via index. Depending on the underlying architecture of the collection, this could get really heavy on ram access and thus decrease performance